healthcheck/healthcheck.go

```diff
@@ -127,11 +130,36 @@ func checkBackend(currentBackend *BackendHealthCheck) {
 	}
 }
 
+func (backend *BackendHealthCheck) newRequest(serverURL *url.URL) *http.Request {
+	// make a copy of the url
+	u := &url.URL{}
+	*u = *serverURL
+
+	if backend.Options.Port != 0 {
+		u.Host = net.JoinHostPort(u.Hostname(), strconv.Itoa(backend.Options.Port))
+	}
+	u.Path = u.Path + backend.Path
+
+	req := &http.Request{
```

Comment:
I'm inclined to say we should just call `http.NewRequest` to be on the safe side, especially if more fields are added over time that might be important for us but would likely be left behind.

WDYT?